pkg/controller/metrics/metrics.go

```diff
@@ -135,6 +141,24 @@ func (mc *Calculator) Start(stopCh <-chan struct{}) error {
 			mcLog.WithField("totalRunningUninstallJobs", uninstallJobsTotal).Debug("loaded running uninstall jobs")
 			metricUninstallJobsRunningTotal.Set(float64(uninstallJobsTotal))
 		}
+		// pod metrics
+		mcLog.Debug("calculating installer pods metrics")
+		pods := &corev1.PodList{}
+		err = mc.Client.List(context.Background(), client.MatchingLabels(installJobLabelSelector), pods)
+		if err != nil {
+			log.WithError(err).Error("error listing pods")
+		} else {
+			podRestarts := 0
+			finalAverage := 0.0
+			for _, pod := range pods.Items {
+				for _, cs := range pod.Status.ContainerStatuses {
+					podRestarts = podRestarts + int(cs.RestartCount)
+				}
+			}
+			finalAverage = float64(podRestarts) / float64(len(pods.Items))
```

Comment:
Looking at: https://jira.coreos.com/browse/CO-293 my thought was this should be in the cluster deployment controller, rather than this metrics controller. I'm still learning about prometheus but for a metric like this I  was thinking that our goal is to record each install job once when it completes. As it stands right now I think we might be re-reporting data, if the pod dies and gets restarted we send a fresh influx of all the existing install jobs, even those that have been reported before. It's hard to know for sure without prometheus to test with, which we don't have right now, but I suspect that's what it must do.

I'm also thinking it's not a gauge, rather a histogram or a summary. (I've tried to document starting ideas in the jira card notes for some of these so keep an eye out) https://prometheus.io/docs/practices/histograms/

Prometheus would have our historical data I think, so our goal here should be to report installer restarts when we detect that the job has completed, one time only. I think this could happen here: https://github.com/openshift/hive/blob/master/pkg/controller/clusterdeployment/clusterdeployment_controller.go#L341 if you were to check if is cd.Status.Installed == false && job.completed = true, we know we've just seen it complete for the first time. However we shouldn't send the metric until we successfully update status below. 

Note that this will also be a good place to record the length of time the job ran for one of these next ones.

Reply:
You shouldn't need to calculate the average either per the above metric type, the histogram/summary link above talks about being used for request times, I think that's essentially what we're doing here, it's just the number of install jobs restarts.